website/addons/github/model.py

```diff
@@ -86,243 +87,101 @@ def _exception_from_response(self, response):
-        if self.oauth_settings:
-            return self.oauth_settings.github_user_name
-        return None
+    client_id = github_settings.GITHUB_CLIENT_ID
+    client_secret = github_settings.GITHUB_CLIENT_SECRET
 
-    @github_user_name.setter
-    def github_user_name(self, user_name):
-        self.oauth_settings.github_user_name = user_name
+    auth_url_base = 'https://github.com/login/oauth/authorize'
+    callback_url = 'https://github.com/login/oauth/access_token'
+    default_scopes = github_settings.SCOPE
 
-    @property
-    def oauth_access_token(self):
-        if self.oauth_settings:
-            return self.oauth_settings.oauth_access_token
-        return None
-
-    @oauth_access_token.setter
-    def oauth_access_token(self, oauth_access_token):
-        self.oauth_settings.oauth_access_token = oauth_access_token
+    _client = None
 
     @property
-    def oauth_token_type(self):
-        if self.oauth_settings:
-            return self.oauth_settings.oauth_token_type
-        return None
+    def api_error_classes(self):
+        return GitHubError
 
-    @oauth_token_type.setter
-    def oauth_token_type(self, oauth_token_type):
-        self.oauth_settings.oauth_token_type = oauth_token_type
+    def handle_callback(self, response):
+        client = self.client(response)
 
-    # Required for importing username from social profile configuration page
-    @property
-    def public_id(self):
-        if self.oauth_settings:
-            return self.oauth_settings.github_user_name
-        return None
-
-    def save(self, *args, **kwargs):
-        if self.oauth_settings:
-            self.oauth_settings.save()
-        return super(AddonGitHubUserSettings, self).save(*args, **kwargs)
-
-    def to_json(self, user):
-        ret = super(AddonGitHubUserSettings, self).to_json(user)
-        ret.update({
-            'authorized': self.has_auth,
-            'authorized_github_user': self.github_user_name if self.github_user_name else '',
-            'show_submit': False,
-        })
-        return ret
-
-    def revoke_token(self):
-        connection = GitHub.from_settings(self)
-        try:
-            connection.revoke_token()
-        except GitHubError as error:
-            if error.code == http.UNAUTHORIZED:
-                return (
-                    'Your GitHub credentials were removed from the OSF, but we '
-                    'were unable to revoke your access token from GitHub. Your '
-                    'GitHub credentials may no longer be valid.'
-                )
-            raise
+        return {
+            'display_name': client.user().name,
+            'provider_id': client.user().id,
+            'profile_url': client.user().html_url,
+        }
 
-    def clear_auth(self, auth=None, save=False):
-        for node_settings in self.addongithubnodesettings__authorized:
-            node_settings.deauthorize(auth=auth, save=True)
+    def client(self, credentials):
```

Comment:
Since the Github class in api.py is such a thin wrapper over gh3, you might consider moving that code into the GithubProvider class.